### DISCUSSION TITLE
Is there a way to replace individual elements in an actionable message?

### DISCUSSION THREAD

**Discussion by surjyams** [microsoft/AdaptiveCards#5328] [Q&A] [answered]
Dear All,
We have a requirement to send actionable messages to our users, similar to an approval email, where a single email contains multiple approval requests. The recipient (approver) is able to approve/reject the requests independently. For example, if the recipient (approver) approves one request out of 5 requests, upon Action.Submit, our API sends back the complete card payload: the payload contains acknowledgement for the approved item, as well as other elements, which have not been touched by the approver: this results in replacing the complete card in the approver's email body.

We would like to update only the particular element, that has been approved, without touching other elements in the body. For one, this will reduce the size of the response payload. Secondly, our API will be able to respond quicker, since it does not have to fetch the status of the other elements, that are not updated.

Thanks in advance, for guiding us here.
~Surjya.

**Comment by surjyams** [comment]
It seems Microsoft uses some undocumented schema to update actionable messages in Outlook. For response to an action on an email from Cortana, the response contains:

```
"status": "Completed",
	"displayMessage": null,
	"hideCard": false,
	"removeCard": false,
	"showOriginalCard": false,
	"isEphemeralRefreshCard": false,
	"cardUpdates": [
		{
			"action": "ReplaceElement",
			"elementId": "5afde905-3bee-44f7-86ec-d8db3c0201b2",
			"element": {
				"width": "auto",
				"items": [
					{
						"altText": "Pinned",
						"pixelWidth": 23,
						"pixelHeight": 23,
						"width": "23px",
						"height": "23px",
						"url": "cid:circle_check_mark.png",
						"horizontalAlignment": "right",
						"type": "Image"
					}
				],
				"verticalContentAlignment": "center",
				"id": "5afde905-3bee-44f7-86ec-d8db3c0201b2",
				"type": "Column",
				"height": "auto"
			}
		}
	],
```

But, searching for tags/keywords in this response did not yield any result. Any pointers, will be extremely helpful.

**Comment by surjyams** [comment]
@matthidinger @dclaux any help please?

**Reply by VMey** [reply]
Not from Microsoft, but the only way you'd be able to do this is by writing a custom element.  There's no way to update a card dynamically currently, because that card represents a point in time of the conversation.

https://docs.microsoft.com/en-us/adaptive-cards/rendering-cards/extensibility

**Comment by dclaux** [comment, marked as answer]
Single element updates are supported in Outlook via Action.Http but not in Teams with Action.Submit. There is currently no plan to add support for it in Teams.
